shm/include/player.hpp

```diff
@@ -0,0 +1,30 @@
+#pragma once
+#include <memory>
+
+#include "ship.hpp"
+#include "time.hpp"
+
+class Player : public Ship::Delegate {
+public:
+    Player(std::shared_ptr<Ship> ship, size_t money);
+    Player(size_t money, std::shared_ptr<Time> time);
+    virtual ~Player() = default;
```

Comment:
= default is definition. Please move to .cpp

Reply:
done